os/pm/pm_changestate.c

```diff
@@ -241,6 +244,11 @@ int pm_changestate(enum pm_state_e newstate)
 		* disagreed and the state has been reverted).  Set the new state.
 		*/
 		pm_changeall(newstate);
+#ifdef CONFIG_PM_METRICS
+		now = clock_systimer();
+		g_pm_metrics_globals.sticks[g_pmglobals.state] += now - g_pm_metrics_globals.stime;
```

Comment:
about sleep / wakeup time,
pm_wakehandler , missing this is better

Please check it more

Reply:
pm_changestate is called inside pm_wakehandler. So we don't need to update there